### PR TITLE
[RORO-183][PH] Initial implementation of create GMR core API

### PR DESCRIPTION
**Pull request by peter-hazell**

### CODE REVIEW

File: app/uk/gov/hmrc/goodsmovementsystemstubs/controllers/NotificationScheduler.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  def scheduleNotification(gmrDocument: GoodsMovementRecordDocument): Unit = {
+    actorSystem.scheduler.scheduleOnce(delay = Random.nextInt(6).seconds) {
+      Logger.info(s"Notification triggered for GMR with ID ${gmrDocument.gmrId}")
+    }
+
+    ()
+  }
+
+}
```
Comment (AKKUM) [inline app/uk/gov/hmrc/goodsmovementsystemstubs/controllers/NotificationScheduler.scala]: Do we need to unit test this class?

File: app/uk/gov/hmrc/goodsmovementsystemstubs/repositories/GoodsMovementSystemRepository.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ) {
+
+  override def ensureIndexes(implicit executionContext: ExecutionContext): Future[Seq[Boolean]] =
+    Future.sequence(
+      Seq(
+        collection.indexesManager.ensure(Index(Seq("gmrId" -> IndexType.Text), Some("GmrId"), unique = true))
+      )
+    )
+
+  def createGmr(gmrDocument: GoodsMovementRecordDocument): Future[GoodsMovementRecordDocument] =
```
Comment (AKKUM) [inline app/uk/gov/hmrc/goodsmovementsystemstubs/repositories/GoodsMovementSystemRepository.scala]: Same as above. Do we need to unit test this class?

Comment (peter-hazell) [reply app/uk/gov/hmrc/goodsmovementsystemstubs/repositories/GoodsMovementSystemRepository.scala]: I don't think we need to test the repository directly in unit tests

I would argue this is tested more effectively through the integration tests

File: app/uk/gov/hmrc/goodsmovementsystemstubs/controllers/NotificationScheduler.scala (hunk 1 above)
Comment (peter-hazell) [reply app/uk/gov/hmrc/goodsmovementsystemstubs/controllers/NotificationScheduler.scala]: Just investigating if there is some nice way I can test the akka scheduler

Comment (peter-hazell) [reply app/uk/gov/hmrc/goodsmovementsystemstubs/controllers/NotificationScheduler.scala]: I moved the logic from the scheduler to it's own service file, as this way any notification sending logic can be tested independently. I don't think we need to have a unit / integration test that proves the akka scheduler itself works.

**Pull request merged**